Keterangan | Isi
-----|-----------------------
Nama | Keisha Prillia Qayyima
NIM  | 16321021
Tanggal | 28/11/2021
Deskripsi | Program menganalisa dataset yang berisi nama, fakultas/sekolah, nilai matematika, fisika dan kimia

In [6]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# read file data (csv)
url = "https://raw.githubusercontent.com/keishapq31/Prak-Pengkom-05/main/H05.csv"
df = pd.read_csv(url)

Program di atas untuk mengimport library yang dibutuhkan dalam program ini secara keseluruhan, serta untuk membaca data yang digunakan.

In [8]:
# 1. Banyaknya data (dimensi dataset)
print("Dimensi dataset: "+str(len(df))+" baris x "+str(len(df.columns))+" kolom.")
print()

Dimensi dataset: 5135 baris x 5 kolom.



In [9]:
# 2. Menunjukkan 10 data pertama
print(df[:10])
print()

                             nama fakultas  nilai_mat  nilai_kim  nilai_fis
0            Farhan Fathur Rahman    FMIPA      57.21      34.14      38.93
1                  Steven Margono    FMIPA      67.43      51.50      57.20
2                    Ryan Sentosa    SAPPK      90.62      37.38      40.73
3            Muhajah Babny Muslim    FMIPA      72.31      38.80      44.79
4                Alvin Natawiguna     FTSL      65.45      57.67      56.39
5                   Sasti Safitri     FTSL      60.58      41.91      51.55
6             Dani Yulianto Putra     FTSL      51.67      35.05      48.69
7  Nabila Febitsukarizky Bunyamin     FTTM      71.66      67.61      69.75
8             Ratna Siti Khodijah      FTI      73.06      59.16      59.48
9                    Gita Partiwi     FTSL      69.57      38.84      48.17



In [10]:
# 3. Menunjukkan data ke 50 sampai 60
print(df[50:61])
print()

                        nama fakultas  nilai_mat  nilai_kim  nilai_fis
50      Arief Syakur Sutedjo     FTTM      84.40      61.21      73.91
51         Fayi Raihan Saleh    FMIPA      60.62      32.57      42.34
52          Erna Kristiawati      FTI      67.48      46.78      52.30
53                    Albert    SAPPK      70.27      50.10      54.01
54  M Jazztyan Indra Pradana     STEI      53.40      43.06      60.30
55      Moch Ginanjar Busiri    FMIPA      74.29      38.44      49.15
56              Aisya Tamara     FITB      71.49      35.86      37.41
57        Tyas Arum Widayati    SAPPK      68.57      37.02      53.25
58                Arif Saleh      FTI     100.00      59.79      55.33
59     Windy Diwita Nisaekti      FTI      67.15      57.72      54.24
60             Alif Aviano S     STEI      54.11      58.01      73.17



In [11]:
# 4. Banyaknya mahasiswa tiap Fakultas
fak = df["fakultas"].value_counts()
print(fak)
print()

FTI      834
STEI     777
FTSL     660
FMIPA    627
FTTM     579
FTMD     480
FITB     403
SAPPK    315
SITHR    282
SITHS    178
Name: fakultas, dtype: int64



In [12]:
# 5. Korelasi nilai Fisika dan Kimia
x = df["nilai_fis"].corr(df["nilai_kim"])
print(x)
print("Dalam persen:",x*100,"%")

0.7988187423880275
Dalam persen: 79.88187423880275 %


Antara nilai Fisika dan Kimia, terdapat korelasi yang cukup besar. Korelasi antara nilai kedua mata kuliah itu mencapai sekitar 79.88%. Korelasi seperti ini menyampaikan bahwa ada hubungan yang cukup kuat antara besar nilai mata kuliah Fisika dan Kimia. Menyatakan bahwa, besar nilai Fisika dan Kimia cenderung berbanding lurus diantara keduanya.